datanode/src/main/java/com/odysseusinc/arachne/datanode/service/client/engine/ExecutionEngineClient.java

```diff
@@ -78,13 +83,18 @@ public AnalysisResultDTO cancel(long id) {
     }
 
     public AnalysisRequestStatusDTO sendAnalysisRequest(
-            AnalysisRequestDTO analysisRequest, boolean compressedResult, String name, RequestBody requestBody
+            AnalysisRequestDTO analysisRequest, boolean compressedResult, String name, IOConsumer<OutputStream> bodyWriter
     ) {
         String json = toJson(analysisRequest);
+        // ZipOutputStream must be auto-closed to finalize the archive, however underlying sink should
+        // not be closed as okhttp might have more data to write in the same request
+        OctetStreamRequestBody x = new OctetStreamRequestBody(sink ->
+                bodyWriter.accept(new NoCloseOutputStream(sink.outputStream()))
+        );
```

Comment:
When is this stream going to be closed? I don't see a specific point in the code where that happens.

Reply:
`NoCloseOutputStream` is just a wrapper around stream provided by `okio.Sink`. This wrapper doesn't capture any resources that need to be released and thus doesn't have to be closed explicitly (just like `ByteArrayOutputStream`).
The wrapped OutputStream is managed by the library and closed when parts of the request are written.